test/integration/manifest_int_test.go

```diff
@@ -50,6 +53,54 @@ func (suite *ManifestIntegrationTestSuite) TestManifest_JSON() {
 	cp.Expect(`"requirements":`)
 }
 
+func (suite *ManifestIntegrationTestSuite) TestManifest_Advanced_Reqs() {
+	suite.OnlyRunForTags(tagsuite.Manifest)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	ts.LoginAsPersistentUser()
+
+	cp := ts.Spawn("config", "set", constants.OptinBuildscriptsConfig, "true")
+	cp.ExpectExitCode(0)
+
+	ts.PrepareProject("ActiveState/cli", "00000000-0000-0000-0000-000000000000")
+	bsf := filepath.Join(ts.Dirs.Work, constants.BuildScriptFileName)
+	fileutils.WriteFile(bsf, []byte(`
+at_time = "2022-07-07T19:51:01.140Z"
+runtime = state_tool_artifacts_v1(src = sources)
+sources = solve(
+	at_time = at_time,
+	requirements = [
+		Req(name = "python", namespace = "language", version = Eq(value = "3.9.13")),
+		Revision(name = "IngWithRevision", revision_id = "00000000-0000-0000-0000-000000000000"),
```

Comment:
Please use a constant for this revision ID, as it's easy to typo it for your tests below.